tests/unit/test_mock_server.py

```diff
@@ -326,6 +335,46 @@ def test_register_handlers_registers_connection_handler_with_bindings_validation
                 handler_with_validation()
 
 
+def test_register_namespace_handlers_emits_security_validator_if_security_enabled():
+    channel_handlers = ChannelHandlers(connect="tests.fixtures.handlers.connect")
+    spec = AsyncApiSpec(
+        asyncapi="2.2.0",
+        info=Info("test", "1.0.0"),
+        servers={
+            "test": Server("https://localhost/", ServerProtocol.WSS, [{"basic": []}])
+        },
+        channels={GLOBAL_NAMESPACE: Channel(x_handlers=channel_handlers)},
+        components=Components(
+            security_schemes={
+                "basic": SecurityScheme(
+                    type=SecuritySchemesType.HTTP,
+                    scheme=HTTPAuthenticationScheme.BASIC,
+                    x_basic_info_func="tests.fixtures.handlers.basic_info",
+                )
+            }
+        ),
+    )
+
+    server = new_mock_asynction_socket_io(spec)
+    server._register_namespace_handlers(
```

Comment:
```suggestion
    server._register_handlers(
```
The Mock server class does not  really implement the `_register_namespace_handlers` method. This success of this test is a false positive. Need to use `_register_handlers` instead.